src/message/renderMessages.js

```diff
@@ -2,49 +2,34 @@
 import React from 'react';
 
 import type { Message, Narrow } from '../types';
-import {
-  isTopicNarrow,
-  isPrivateOrGroupNarrow,
-  extractTypeToAndSubjectFromNarrow,
-} from '../utils/narrow';
+import { isTopicNarrow, isPrivateOrGroupNarrow } from '../utils/narrow';
 import MessageHeaderContainer from '../message/headers/MessageHeaderContainer';
 import MessageContainer from '../message/MessageContainer';
-import OutboxMessageContainer from '../message/OutboxMessageContainer';
 import TimeRow from '../message/TimeRow';
 import { isSameRecipient } from '../utils/message';
 import { isSameDay } from '../utils/date';
 
 export default (messages: Message[], narrow: Narrow) => {
   const list: Object[] = [];
   let prevItem;
+  const showHeader = !isPrivateOrGroupNarrow(narrow) && !isTopicNarrow(narrow);
 
   for (const item of messages) {
-    const isOutbox = !item.id;
+    const isOutbox = 'isOutbox' in item && item.isOutbox === true;
```

Comment:
You should create normal looking messages in the selector. This function should not be aware of outbox at all.